src/compiler/visitorPublic.ts

```diff
@@ -36,7 +36,7 @@ namespace ts {
             return undefined;
         }
         else if (isArray(visited)) {
-            visitedNode = (lift || extractSingleNode)(visited);
+            visitedNode = (lift || extractSingleNode)(visited as unknown as NodeArray<Node>);

```

Comment:
cc @weswigham @ahejlsberg interesting break here. `visited` is narrowed to `Node[]` here, but `lift` is `undefined | (arg: NodeArray<Node>) => Node`, so we were passing what we only proved to be a `Node[]` to something that needed a subtype of `Node[]`. It's not immediately clear to me if this is a bug-in-fact but it's definitely not correct per our normal definitions, i.e. you could not have written
```ts
if (lift) lift(visited);
```
on this line